iceoryx_binding_c/include/iceoryx_binding_c/types.h

```diff
@@ -18,8 +18,21 @@
 #ifndef IOX_BINDING_C_TYPES_H
 #define IOX_BINDING_C_TYPES_H
 
+#include "iceoryx_binding_c/iceoryx_binding_c_deployment.h"
 #include "internal/c2cpp_binding.h"
 
+#if defined(__APPLE__)
+#define CALCULATE_STORAGE_SIZE_FOR_LISTENER(numberOfAttachments)                                                       \
+    (144 + numberOfAttachments * 168 - 8 * (((numberOfAttachments + 1) / 2) - 1))
+#elif defined(_WIN32)
+#define CALCULATE_STORAGE_SIZE_FOR_LISTENER(numberOfAttachments)                                                       \
+    (168 + numberOfAttachments * 176 - 8 * (((numberOfAttachments + 1) / 2) - 1))
+#else
+#define CALCULATE_STORAGE_SIZE_FOR_LISTENER(numberOfAttachments) (((128 + numberOfAttachments * 140) / 8) * 8)
+#endif
+
+#define CALCULATE_STORAGE_SIZE_FOR_WAITSET(numberOfAttachments) (552 + numberOfAttachments * 184)
```

Comment:
Counter proposal to this mess #1169

Reply:
I find the proposal is the right way to go, but not for 2.0 - not enough time and it is uncertain that we encounter some weird surprises here.

We have tests in place to handle it cleanly and provide a working solution for the user so I would pursue this approach for now.